### PR TITLE
chore(core): add schema definition test utility

### PR DESCRIPTION
**Pull request by fuchsnj**
`schema::Definition::assert_valid_for_event` was added as an easy way to validate that a schema definition is valid for a specific event. This can easily be added to integration tests, or any test where you already have valid events, and the associated schema definition.

`assert_valid_for_event` will panic is the event doesn't match, and print out debug information.
`is_valid_for_event` is available if you don't want a panic. It will still give the debug info as a `String`.

Both of these functions are only available in tests. They are available in `vector-core` when running tests, or in any other crate by enabling the `test` feature of `vector-core` in `dev-dependencies` (which many crates do already).

### CODE REVIEW

File: lib/vector-core/src/schema/definition.rs [hunk 1]
```diff
@@ -452,13 +452,120 @@ impl Definition {
...
 }
 
+#[cfg(any(test, feature = "test"))]
+mod test_utils {
+    use super::*;
+    use crate::event::{Event, LogEvent};
+
+    impl Definition {
+        /// Asserts that the schema definition is _valid_ for the given event.
+        /// This will panic if the definition is not valid.
```
Comment (tobz) [inline lib/vector-core/src/schema/definition.rs:462]: Doesn't look like it panics?

File: lib/vector-core/src/event/metadata.rs [hunk 2]
```diff
@@ -43,6 +43,14 @@ fn default_metadata_value() -> Value {
 
 /// Secret access functions
```
Comment (spencergilbert) [inline lib/vector-core/src/event/metadata.rs:44]: 👀 this isn't _just_ secrets anymore yeah?

File: lib/vector-core/src/schema/definition.rs (hunk 1 above)
Comment (spencergilbert) [reply lib/vector-core/src/schema/definition.rs:462]: I guess this should be down with the `assert` fn?

**Review by StephenWakely — APPROVED** [review]
This is working beautifully for me!

**Pull request merged**